src/modules/cpu/kernel/normalize.hpp

```diff
@@ -505,6 +572,134 @@ void collapse_axis(Rpp32u *tensorDim, Rpp32u *axis, Rpp32u *length, Rpp32u *newA
+    Rpp32u batchSize = dstGenericDescPtr->dims[0];
+
+    Rpp32u maxSize = 1;
+    // Compute maxSize as length of input tensors differ based on axisMask and tensorDims
+    for(int batch = 0; batch < batchSize; batch++)
+    {
+        Rpp32u size = 1;
+        for(int i = 0; i < tensorDims; i++)
+            size *= ((axisMask & (int)(pow(2, i))) >= 1) ? 1 : roiTensor[(tensorDims * 2 * batch) + tensorDims + i];
+        maxSize = std::max(maxSize, size);
+    }
+
+    if(!computeMeanStddev)
+    {
+        for(Rpp32u i = 0; i < maxSize; i++)
+            stdDevTensorPtr[i] = (!stdDevTensorPtr[i])? 1.0f : scale / stdDevTensorPtr[i];
+        maxSize = 0;
+    }
+
+        omp_set_dynamic(0);
+#pragma omp parallel for num_threads(numThreads)
+    for(int batchCount = 0; batchCount < batchSize; batchCount++)
+	{
+        Rpp32u *roi = roiTensor + batchCount * tensorDims * 2;
+        Rpp32u *begin = roi;
+        Rpp32u *length = &roi[tensorDims];
+        Rpp8u *srcPtrTemp, *dstPtrTemp; 
+        Rpp32f *meanTensor, *stdDevTensor;
+        srcPtrTemp = srcPtr + batchCount * srcGenericDescPtr->strides[0];
+        dstPtrTemp = dstPtr + batchCount * dstGenericDescPtr->strides[0];
+        meanTensor = meanTensorPtr + batchCount * maxSize;
+        stdDevTensor = stdDevTensorPtr + batchCount * maxSize;
+        // Set all values in dst buffer to 0.0
+        for(int cnt = 0; cnt < dstGenericDescPtr->strides[0]; cnt++)
+            dstPtrTemp[cnt] = 0;
+
+        Rpp8u *srcPtrChannel = srcPtrTemp;
+
+        if(tensorDims == 1)
+        {
+            Rpp32u srcReductionDims, srcStride;
+            srcReductionDims = length[0];
+            srcStride = srcGenericDescPtr->strides[1];
+            Rpp32f normFactor = (Rpp32f)(1.0 / srcReductionDims);
+            Rpp32f *meanPtr = meanTensor;
+            Rpp32f *stdDevPtr = stdDevTensor;
+            if(computeMeanStddev & 1) // Check if mean is to be computed internally
+            {
+                meanPtr[0] = 0;
+                compute_sum(meanPtr[0], srcPtrTemp, srcStride, srcReductionDims);
+                meanPtr[0] *= normFactor;
+            }
+            if(computeMeanStddev & 2) // Check if stddev is to be computed internally
+            {    
+                stdDevPtr[0] = 0;
+                compute_diff_square_sum(stdDevPtr[0], srcPtrTemp, srcStride, srcReductionDims, meanPtr[0]);
+                rpp_rsqrt_sse(stdDevPtr, 1, 0, normFactor, scale);
+            }
+            normalize_1D_tensor(srcPtrTemp, srcGenericDescPtr, dstPtrTemp, dstGenericDescPtr, meanPtr, stdDevPtr, shift, length);
+
+        }
+        else
+        {
+            int size = 1;
+            for(int i = 0; i < tensorDims; i++)
+                size *= ((axisMask & (int)(pow(2, i))) >= 1) ? 1 : length[i];
+            
+            Rpp32u totalElements = 1;
+            Rpp32u lastNormAxis = 0;
+            Rpp32u axis[tensorDims], newAxis[tensorDims], newDims[tensorDims];
```

Comment:
you are doing static allocations here. This might cause some issue . please use the scratch buffer available in the host handle